### ISSUE TITLE
Enabling the monitor layer while running vkreplay causes an undefined symbol error

### ISSUE THREAD

**Issue by johnzupin** [LunarG/VulkanTools#801]
To replicate:
Download the SDK, setup your environment variables, then create a vktrace of vkcube. Replay the vktrace with the monitor layer enabled like so:
`VK_INSTANCE_LAYERS=VK_LAYER_LUNARG_monitor vkreplay -o vkcube_trace.vktrace`

Here is the error:
```
johnz@neon:~/1.1.106.0/examples/build VK_INSTANCE_LAYERS=VK_LAYER_LUNARG_monitor vkreplay -o vkcube_trace.vktrace 
vkreplay: symbol lookup error: /home/johnz/1.1.106.0/x86_64/lib/libVkLayer_monitor.so: undefined symbol: xcb_get_property
```
If this layer is enabled by itself or with other layers, it will cause this error with vkreplay.

**Comment by Rua** [comment]
I get the same error when enabling the monitor layer. Without it, everything is fine.

**Comment by Rua** [comment]
Judging from the source code, it seems that the monitor layer works only with xcb, not with xlib. In my case, I'm indirectly using xlib. In the meantime before xlib support is implemented, the library could at least try to detect that the application isn't using xcb and disable the layer.

**Comment by ROTARTSI82** [comment]
Is it possible to make the error message more clear or even prevent `VK_LAYER_LUNARG_monitor` from showing up in `vkEnumerateInstanceExtensionProperties` in the first place if it isn't supported?

**Comment by davidlunarg** [comment]
We realize this is an issue and plan to provide a better/improved monitor layer in the future. For now, this issue is lower priority than other work. But we would certainly accept a pull request. :-)

**Comment by KarenGhavam-lunarG** [comment]
@davidlunarg Didn't you recently fix this in PR #1004 ?

**Comment by davidlunarg** [comment]
We believe this has been fixed in commit 91cfa3115a6ef7f76feb5f3e07e17203c71f8ea8. However, since vkreplay has been deprecated, the fix has not been verified with vkreplay. The fix was only verified with gfxrecon-replay.
